pvlib/test/test_singlediode_methods.py

```diff
@@ -125,3 +126,135 @@ def test_brentq_fs_495():
+    Vd = fs_495['I_sc_ref'] * fs_495['R_s']  # diode voltage at short circuit
+    Id = fs_495['I_o_ref'] * (np.exp(Vd / nNsVt) - 1)  # diode current (A)
+    Ish = Vd / fs_495['R_sh_ref']  # shunt current (A)
+    # builtin potential difference (V)
+    dv = VOLTAGE_BUILTIN * fs_495['cells_in_series'] - Vd
+    # calculate photo-generated current at reference condition (A)
+    fs_495['I_L_ref'] = (
+        (fs_495['I_sc_ref'] + Id + Ish) / (1 - fs_495['d2mutau'] / dv)
+    )
+    return fs_495
+
+
+PVSYST_FS_495 = pvsyst_fs_495()  # PVSyst First Solar FS-495 parameters
+
+
+def test_pvsyst_fs_495_recombination_loss():
+    """test pvsystem.singlediode with Brent method on SPR-E20-327"""
+    poa, temp_cell = 1000.0, 25.0  # test conditions
+
+    # first evaluate DeSoto model
+    cec_fs_495 = CECMOD.First_Solar_FS_495  # CEC parameters for
+    il_cec, io_cec, rs_cec, rsh_cec, nnsvt_cec = pvsystem.calcparams_desoto(
+        effective_irradiance=poa, temp_cell=temp_cell,
+        alpha_sc=cec_fs_495.alpha_sc, a_ref=cec_fs_495.a_ref,
+        I_L_ref=cec_fs_495.I_L_ref, I_o_ref=cec_fs_495.I_o_ref,
+        R_sh_ref=cec_fs_495.R_sh_ref, R_s=cec_fs_495.R_s,
+        EgRef=1.475, dEgdT=-0.0003
+    )
+    voc_est_cec = estimate_voc(photocurrent=il_cec, saturation_current=io_cec,
+                               nNsVth=nnsvt_cec)
+    vd_cec = np.linspace(0, voc_est_cec, 1000)
+    desoto = bishop88(
+        diode_voltage=vd_cec, photocurrent=il_cec, saturation_current=io_cec,
+        resistance_series=rs_cec, resistance_shunt=rsh_cec, nNsVth=nnsvt_cec
+    )
+
+    # now evaluate PVSyst model with thin-film recombination loss current
+    pvsyst_fs_495 = PVSYST_FS_495
+    x = pvsystem.calcparams_pvsyst(
+        effective_irradiance=poa, temp_cell=temp_cell,
+        alpha_sc=pvsyst_fs_495['alpha_sc'],
+        gamma_ref=pvsyst_fs_495['gamma_ref'],
+        mu_gamma=pvsyst_fs_495['mu_gamma'], I_L_ref=pvsyst_fs_495['I_L_ref'],
+        I_o_ref=pvsyst_fs_495['I_o_ref'], R_sh_ref=pvsyst_fs_495['R_sh_ref'],
+        R_sh_0=pvsyst_fs_495['R_sh_0'], R_sh_exp=pvsyst_fs_495['R_sh_exp'],
+        R_s=pvsyst_fs_495['R_s'],
+        cells_in_series=pvsyst_fs_495['cells_in_series'],
+        EgRef=pvsyst_fs_495['EgRef']
+    )
+    il_pvsyst, io_pvsyst, rs_pvsyst, rsh_pvsyst, nnsvt_pvsyst = x
+    voc_est_pvsyst = estimate_voc(photocurrent=il_pvsyst,
+                                  saturation_current=io_pvsyst,
+                                  nNsVth=nnsvt_pvsyst)
+    vd_pvsyst = np.linspace(0, voc_est_pvsyst, 1000)
+    pvsyst = bishop88(
+        diode_voltage=vd_pvsyst, photocurrent=il_pvsyst,
+        saturation_current=io_pvsyst, resistance_series=rs_pvsyst,
+        resistance_shunt=rsh_pvsyst, nNsVth=nnsvt_pvsyst,
+        d2mutau=pvsyst_fs_495['d2mutau'],
+        cells_in_series=pvsyst_fs_495['cells_in_series']
+    )
+
+    # test expected change in max power
+    assert np.isclose(max(desoto[2]) - max(pvsyst[2]), 0.01949420697212645)
```

Comment:
This test passes if the Desoto model and the PVsyst model are 'close', meaning that the test would fail is the CEC parameter database entry for this module changes, if there is an error in the Desoto functions, or if the PVsyst function has an error.  I'd change the test condition to compare the PVsyst output to known values calculated by a separate method.

Reply:
✔️ I've changed the test to be independent of DeSoto, now it only tests 2 conditions:
* reference condition, should agree with its own parameters
* another condition, (888[W/m^2], 55[degC]), for which I calculated the expected output using `calcparams_pvsyst` and `bishop88`


But I should note that the output at (888[W/m^2], 55[degC]) doesn't agree well with DeSoto at the same conditions:

```python
In [1]: from pvlib import pvsystem
In [2]: CECMOD = pvsystem.retrieve_sam('cecmod')
In [3]: fs_495 = CECMOD.First_Solar_FS_495
In [4]: il, io, rs, rsh, nnsvt= pvsystem.calcparams_desoto(888, 55,
   ...: alpha_sc=fs_495.alpha_sc, a_ref=fs_495.a_ref, I_L_ref=fs_495.I_L_ref,
   ...: I_o_ref=fs_495.I_o_ref, R_sh_ref=fs_495.R_sh_ref, R_s=fs_495.R_s,
   ...: EgRef=1.475, dEgdT=-0.0003)
In [5]: pvs = pvsystem.singlediode(il, io, rs, rsh, nnsvt, method='lambertw')
In [6]: pvs
Out[6]:
OrderedDict([('i_sc', 1.4020527838304147),  # PVSyst: 1.3868344548308347
             ('v_oc', 75.41704855182024),  # PVSyst: 79.29198489135703
             ('i_mp', 1.2599626119945482),
             ('v_mp', 57.70813302772382),
             ('p_mp', 72.71009002293975),  # PVSyst: 76.26211540956041 
             ('i_x', 1.3606253515830833),
             ('i_xx', 0.8340096657730403)])
```

According to @jdnewmil this is possibly due to a setting in PVSyst that adjusts `d2mutau` to a fixed max power temp coeff instead (or in addition???) to the adjustment on `gamma`. Even though I can see these check boxes in PVSyst, I'm not sure to what degree we want to copy or guess what PVSyst does here in PVLib (or in SAM either @timorichert)

Reply:
@cwhanse if we wanted to try to emulate this 2nd order `d2mutau` behavior, then IMO it should go into `calcparams_pvsyst`, and so those values: `d2mutau` (and `voltage_builtin` if multi junction?) should be outputs from there.

Reply:
Agree. I don't know what PVsyst might be doing behind the interface when `d2mutau` is in action, however, so I can't help clarify here.

Reply:
If it isn't clear what PVsyst is doing, then I suggest mailing Bruno to get clarification.  I think they're quite open about the calculations.